### DISCUSSION TITLE
Convert ONNX model to OpenVINO

### DISCUSSION THREAD

**Discussion by ShidiDaisy** [onnx/onnx#3003] [Q&A]
# Ask a Question

### Question
Hi, I'm trying to use OpenVINO toolkit to convert an ONNX pretrained model from [here](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/faster-rcnn/model/FasterRCNN-10.onnx). Steps are from [here](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_ONNX.html)
But when I ran the command

```
sudo python3 mo.py --input_model FasterRCNN-10.onnx
```

I got these issues:

```
Model Optimizer arguments:
Common parameters:
	- Path to the Input Model: 	/opt/intel/openvino_2020.4.287/deployment_tools/model_optimizer/FasterRCNN-10.onnx
	- Path for generated IR: 	/opt/intel/openvino_2020.4.287/deployment_tools/model_optimizer/.
	- IR output name: 	FasterRCNN-10
	- Log level: 	ERROR
	- Batch: 	Not specified, inherited from the model
	- Input layers: 	Not specified, inherited from the model
	- Output layers: 	Not specified, inherited from the model
	- Input shapes: 	Not specified, inherited from the model
	- Mean values: 	Not specified
	- Scale values: 	Not specified
	- Scale factor: 	Not specified
	- Precision of IR: 	FP32
	- Enable fusing: 	True
	- Enable grouped convolutions fusing: 	True
	- Move mean values to preprocess section: 	False
	- Reverse input channels: 	False
ONNX specific parameters:
Model Optimizer version: 	
[ ERROR ]  Shape [ 1 64 -1 -1] is not fully defined for output 0 of "5". Use --input_shape with positive integers to override model input shapes.
[ ERROR ]  Cannot infer shapes or values for node "5".
[ ERROR ]  Not all output shapes were inferred or fully defined for node "5". 
 For more information please refer to Model Optimizer FAQ (https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html), question #40. 
[ ERROR ]  
[ ERROR ]  It can happen due to bug in custom shape infer function <function Elementwise.__init__.<locals>.<lambda> at 0x7f65a6129e18>.
[ ERROR ]  Or because the node inputs have incorrect values/shapes.
[ ERROR ]  Or because input shapes are incorrect (embedded to the model or passed via --input_shape).
[ ERROR ]  Run Model Optimizer with --log_level=DEBUG for more information.
[ ERROR ]  Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.middle.PartialInfer.PartialInfer'>): Stopped shape/value propagation at "5" node. 
 For more information please refer to Model Optimizer FAQ (https://docs.openvinotoolkit.org/latest/_docs_MO_DG_prepare_model_Model_Optimizer_FAQ.html), question #38. 
```

Looks like the input shape in the model is invalid? How should I solve this issue? What input shape should I use to overwrite the model input shape?

### Further information
- Relevant Area: shape_inference

- Is this issue related to a specific model?  
**Model name** (*e.g. mnist*):  [Faster R-CNN](https://github.com/onnx/models/blob/master/vision/object_detection_segmentation/faster-rcnn/model/FasterRCNN-10.onnx)
**Model opset** (*e.g. 7*):

### Notes
Any additional information, code snippets.

**Comment by jcwchen** [comment]
Hi @ShidiDaisy,
It shouldn't be an `onnx.shape_inference` issue. This model can be normally `shape_inference` by onnx 1.7.0. I am not familiar with OpenVINO, but I would say this should be related to their end and you might provide more information to the API like input_shape.

**Comment by ShidiDaisy** [comment]
> Hi @ShidiDaisy,
> It shouldn't be an `onnx.shape_inference` issue. This model can be normally `shape_inference` by onnx 1.7.0. I am not familiar with OpenVINO, but I would say this should be related to their end and you might provide more information to the API like input_shape.

Thanks for your comment.
I have also tried to include --input shape in command, but for this onnx model, the input shape is 3 x height x weight, which cannot be parsed by OpenVINO. I will also try to ask OpenVINO side.
